hack/vendor-console.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -e
+
+STARTTIME=$(date +%s)
+OS_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${OS_ROOT}/hack/common.sh"
+
+CONSOLE_REPO_PATH=${CONSOLE_REPO_PATH:-$OS_ROOT/../origin-web-console}
+if [ ! -d "$CONSOLE_REPO_PATH" ]; then
+  echo "The console repo at path $CONSOLE_REPO_PATH does not exist."
+  echo "Make sure you have cloned the web console repo locally:  git@github.com:openshift/origin-web-console.git"
+  echo "Or, you can specify a different path with the CONSOLE_REPO_PATH environment variable."
+  exit 1
+fi
+
+echo "Making sure go-bindata binary has been built..."
+pushd ${OS_ROOT}/Godeps/_workspace > /dev/null
+  godep_path=$(pwd)
+  pushd src/github.com/jteeuwen/go-bindata > /dev/null
+    GOPATH=$godep_path go install ./...
+  popd > /dev/null
+popd > /dev/null
+
+if [ -z ${GIT_REF+x} ]; then 
+  echo "No git ref specified, using current state of the repo including any unstaged changes...";
+else 
+  echo "Using git ref ${GIT_REF}, all untracked and uncommitted changes will be stashed..."
+  # TODO improve this to return the user back to the state they were in
+  pushd "${CONSOLE_REPO_PATH}" > /dev/null
+    git add .
+    git stash
```

Comment:
See my TODO above :)  there are lot of things we should do, @liggitt has
some pretty amazing utils for this which we could use, this is step 1 for
now so we can get to where QE can test things

On Fri, May 13, 2016 at 12:55 PM, Sam Padgett <notifications@github.com>
wrote:

> In hack/vendor-console.sh
> <https://github.com/openshift/origin/pull/8875#discussion_r63215605>:
>
> > +echo "Making sure go-bindata binary has been built..."
> > +pushd ${OS_ROOT}/Godeps/_workspace > /dev/null
> > +  godep_path=$(pwd)
> > +  pushd src/github.com/jteeuwen/go-bindata > /dev/null
> > +    GOPATH=$godep_path go install ./...
> > +  popd > /dev/null
> > +popd > /dev/null
> > +
> > +if [ -z ${GIT_REF+x} ]; then
> > +  echo "No git ref specified, using current state of the repo including any unstaged changes...";
> > +else
> > +  echo "Using git ref ${GIT_REF}, all untracked and uncommitted changes will be stashed..."
> > +  # TODO improve this to return the user back to the state they were in
> > +  pushd "${CONSOLE_REPO_PATH}" > /dev/null
> > +    git add .
> > +    git stash
>
> Should we git stash pop when done?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8875/files/594c625e45b13e8f6c1748a48d1ed387b3a6777c#r63215605>
>